modules/mpi/tutorials/ospMPIDistribTutorialVolumeHDF5.cpp

```diff
@@ -90,13 +90,61 @@ int main(int argc, char **argv)
         },
         nullptr);
 
+    std::cerr << "ABOUT TO READ" << std::endl;
+
     // open the hdf5 file, read the data and create the ospray Volume object
     ChomboHDF5::Reader hdf5reader(argv[1], mpiRank, mpiWorldSize, argv[2]);
+    std::cerr << "READ!" << std::endl;
+
+    VolumeBrick brick;
+
+    //convert from int to float as required by ospray
+    //get the world bounds while we are at it
+    std::vector<box3f> myregions;
+    vec3f min{MAXFLOAT};
+    vec3f max{-MAXFLOAT};
+    for (auto x : hdf5reader.m_blockBounds) {
+        std::cerr << x.lower << " " << x.upper << std::endl;
+        if (x.lower.x < min.x)
+            min.x = x.lower.x;
+        if (x.lower.y < min.y)
+            min.y = x.lower.y;
+        if (x.lower.z < min.z)
+            min.z = x.lower.z;
+        if (x.lower.x > max.x)
+            max.x = x.lower.x;
+        if (x.lower.y > max.y)
+            max.y = x.lower.y;
+        if (x.lower.z > max.z)
+            max.z = x.lower.z;
+        myregions.push_back(box3f(x.lower, x.upper));
+    }
+    //std::cerr << min << " " << max << std::endl;
+
+    brick.brick = hdf5reader.getVolume();
+    brick.bounds = myregions;
+
+    worldBounds = box3f(min*0.25, max*0.25); //seems like we are missing the block scale somewhere
+
+    brick.model = cpp::VolumetricModel(brick.brick);
+    cpp::TransferFunction tfn("piecewiseLinear");
+    std::vector<vec3f> colors = {vec3f(0.f, 0.f, 1.f), vec3f(1.f, 0.f, 0.f)};
+    std::vector<float> opacities = {0.0f, 0.5f, 1.0f};
+    tfn.setParam("color", cpp::CopiedData(colors));
+    tfn.setParam("opacity", cpp::CopiedData(opacities));
+    vec2f valueRange = vec2f(0, 4080.02);
```

Comment:
Is it worth adding a feature in the reader to determine the extremal values for the read in data?

Reply:
yes, because of the opaque API, it is harder than I'd want to query the ranges after we make the underlying array. So yes please do add it to the reader. Make it optional if you like as an optimization path when we happen to know the range already.

Reply:
Done. It's on my branch as `ChomboHDF5::Reader::calculateValueRange()`.